EmeraldWarden_main.lua

```diff
@@ -0,0 +1,342 @@
+	local unitTarget = behaviorLib.heroTarget 
+	
+	local bActionTaken = false
+	
+	local nDistSq = Vector3.Distance2DSq(unitSelf:GetPosition(), unitTarget:GetPosition())
+	local bTargetMagicImmune = unitTarget:IsMagicImmune()
+	local vecTargetPosition = unitTarget:GetPosition()
+	
+	
+	
+	--since we are using an old pointer, ensure we can still see the target for entity targeting
+	if unitTarget ~= nil then 				
+		
+		-- Since all of Emerald Warden's skills are Magic Dmg, EW will use skill only when his target is NOT MAGIC IMMUNE
+		
+		-- Item: Symbol Of Rage -> Activates when EW is below 60% hp
+		local bHealthLow = unitSelf:GetHealthPercent() < 0.60
+		if not bActionTaken  and bHealthLow then
+			local itemSymbol = core.GetItem("Item_LifeSteal4")
+			if itemSymbol then
+				if itemSymbol:CanActivate() then
+						bActionTaken = core.OrderItem(botBrain, itemSymbol)
+						behaviorLib.lastHarassUtil = behaviorLib.lastHarassUtil + 20   --  Increases aggresion so EW will activate Sybmol and attack not run
+				end
+			end
+		end
+		
+		-- 1st Skill -> Silencing Shot on channeling target if target is NOT MAGIC IMMUNE
+		if unitTarget:IsChanneling() and not bTargetMagicImmune  then
+			local abilSilence = skills.abilSilence
+			if abilSilence:CanActivate() then
+				core.OrderAbilityEntity(botBrain, abilSilence, unitTarget)
+			end
+		end
+		
+		-- 2nd Skill -> Hunter's Command
+		if not bTargetMagicImmune then
+			if not bActionTaken then
+				local abilWolves = skills.abilWolves
+				if abilWolves:CanActivate() and behaviorLib.lastHarassUtil > object.nWolvesThreshold then
+					bActionTaken = core.OrderAbility(botBrain, abilWolves)
+				end
+			end
+		end
+		
+		-- 1st Skill -> Silencing Shot
+		if not bTargetMagicImmune then
+			if not bActionTaken then
+				local abilSilence = skills.abilSilence
+				if abilSilence:CanActivate() and behaviorLib.lastHarassUtil > object.nSilenceThreshold then
+					bActionTaken = core.OrderAbilityEntity(botBrain, abilSilence, unitTarget)
+				end
+			end
+		end
+		
+		-- 3rd Skill -> Overgrowth
+		if not bTargetMagicImmune then
+			if not bActionTaken then
+				local abilOvergrowth = skills.abilOvergrowth
+				if abilOvergrowth:CanActivate() and behaviorLib.lastHarassUtil > object.nOvergrowthThreshold then
+					bActionTaken = core.OrderAbilityPosition(botBrain, abilOvergrowth, vecTargetPosition)
+				end
+			end
+		end
+		
+		
+		
+	end
+	
+	
+
+	
+	if not bActionTaken then
+		return object.harassExecuteOld(botBrain)
+	end
+end
+object.harassExecuteOld = behaviorLib.HarassHeroBehavior["Execute"]
+behaviorLib.HarassHeroBehavior["Execute"] = HarassHeroExecuteOverride
+
+-----------------------
+-- Return to well
+-----------------------
+
+-- Escape combo for running away
+
+function behaviorLib.CustomRetreatExecute(botBrain)
+	bActionTaken = false
+
+	local unitSelf = core.unitSelf
+	local unitSelfPosition = unitSelf:GetPosition()
+	local bCanSee = unitTarget and core.CanSeeUnit(botBrain, unitTarget)
```

Comment:
unitTarget is nil here. You need to pick target from core.localUnits["EnemyHeroes"]